core/src/main/scala/org/apache/spark/deploy/SparkHadoopUtil.scala

```diff
@@ -102,11 +102,19 @@ class SparkHadoopUtil extends Logging {
         hadoopConf.set("fs.s3n.awsSecretAccessKey", accessKey)
         hadoopConf.set("fs.s3a.secret.key", accessKey)
       }
-      // Copy any "spark.hadoop.foo=bar" system properties into conf as "foo=bar"
       conf.getAll.foreach { case (key, value) =>
+        // Copy any "spark.hadoop.foo=bar" system properties into conf as "foo=bar"
         if (key.startsWith("spark.hadoop.")) {
           hadoopConf.set(key.substring("spark.hadoop.".length), value)
         }
+	// Copy any "fs.swift2d.foo=bar" properties into conf as "fs.swift2d.foo=bar"
+        else if (key.startsWith("fs.swift2d")){
+          hadoopConf.set(key, value)
```

Comment:
What's `swift2d`? It's not the swift client in `hadoop-openstack`, which is `fs.swift`